_test/bootstrap.sh

```diff
@@ -1,27 +1,25 @@
 #!/bin/bash
 # This installs a the prerequisites for a given GHC/Cabal config
-set -x
+#set -x
 set -e
 if [ ! -z "$GHCVER" ]; then
     export PATH=/opt/ghc/${GHCVER}/bin:$PATH
 fi
 if [ ! -z "$CABALVER" ]; then
     export PATH=/opt/cabal/${CABALVER}/bin:$PATH
 fi
+
+# What directory is this script in?
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+deps=$(cut -d: -f2 "$DIR/dependencies.txt" | xargs echo | tr " " "\n" | sort | uniq)
+
 cabal update
-# This is a fairly minimal set
-cabal install \
-      primitive \
-      random \
-      tf-random \
-      HUnit \
-      QuickCheck \
-      split \
-      text \
-      bytestring \
-      attoparsec \
-      vector \
-      parallel \
-      stm \
-      old-locale \
-      lens
+
+for dep in $deps; do
+  # Don't update packages included in GHC
+  if [ "$dep" = "base" -o "$dep" = "array" -o "$dep" = "containers" -o "$dep" = "time" ]; then
```

Comment:
this is a "wat". how can we improve this? I would not like have to hard-code these.